library/packages/src/modules/PackageSystem.rb

```diff
@@ -335,22 +337,23 @@ def InitRPMQueryBinary
     end
 
     # Is a package provided in the system? Is there any installed package providing 'package'?
-    # @return true if yes
+    #
+    # @param package [String] name of the package to check if provided
+    # @return [Boolean] whether the package is provided in the system or not
     def Installed(package)
       # This is a most commonly called function and so it's
       # important that it's fast, especially in the common
       # case, where all dependencies are satisfied.
       # Unfortunately, initializing Pkg reads the RPM database...
       # so we must avoid it.
       # added --whatprovides due to bug #76181
-      0 ==
-        Convert.to_integer(
-          SCR.Execute(
-            path(".target.bash"),
-            Ops.add("/usr/bin/rpm -q --whatprovides ", package.shellescape)
-          )
-        )
+      rpm_command = "/usr/bin/rpm -q --whatprovides #{package.shellescape}"
+      log.debug("Starting RPM query: #{rpm_command}")
+      output = SCR.Execute(path(".target.bash_output"), rpm_command)
+      log.debug("result of the query: #{output}")
```

Comment:
well, I would make it non-debug ( info ) and also compress in into one line (rpm -q does not take so long and it is already logged in debug level) like:
```
log.info "Query installed package with '#{rpm_command}' and result #{output.inspect}"
```

Reply:
I was not sure about loging always.